From 4a40a5e603af414bd20a5b1a1f1a7144f8f459f7 Mon Sep 17 00:00:00 2001
From: Jian He <jianhe@apache.org>
Date: Thu, 2 Mar 2017 10:58:56 -0800
Subject: [PATCH 2377/2748] MAPREDUCE-6852. Job#updateStatus() failed with NPE
 due to race condition. Contributed by Junping Du

(cherry picked from commit 747bafaf969857b66233a8b4660590bdd712ed7d)

Change-Id: I0d2d10db760b2c9e825e3f4490b86262d5c1d882
---
 .../main/java/org/apache/hadoop/mapreduce/Job.java |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapreduce/Job.java b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapreduce/Job.java
index 16fc42d..8353a5c 100644
--- a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapreduce/Job.java
+++ b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapreduce/Job.java
@@ -323,7 +323,7 @@ synchronized void updateStatus() throws IOException {
       this.status = ugi.doAs(new PrivilegedExceptionAction<JobStatus>() {
         @Override
         public JobStatus run() throws IOException, InterruptedException {
-          return cluster.getClient().getJobStatus(status.getJobID());
+          return cluster.getClient().getJobStatus(getJobID());
         }
       });
     }
-- 
1.7.9.5

